tests/test_speed.py

```diff
@@ -137,190 +163,107 @@ async def test_class_creation_speed(self):
-                for i in range(num_commands):
-                    await remote.cmd_fault.start(timeout=STD_TIMEOUT)
-                dt = time.monotonic() - t0
-                command_speed = num_commands / dt
-                print(f"issued {command_speed:0.0f} fault commands/second")
-
-                if verify is not None:
-                    self.insert_measurement(
-                        verify.Measurement(
-                            "salobj.IssueCommands", command_speed * u.ct / u.second
-                        )
+            if verify is not None:
+                self.insert_measurement(
+                    verify.Measurement(
+                        "salobj.IssueCommands", command_speed * u.ct / u.second
                     )
-
-        finally:
-            if process.returncode is None:
-                process.kill()
+                )
 
     async def test_read_speed(self):
-        script_path = self.datadir / "topic_writer.py"
-        process = await asyncio.create_subprocess_exec(
-            str(script_path), str(self.index)
-        )
-        try:
-            async with salobj.Domain() as domain, salobj.Remote(
-                domain=domain, name="Test", index=self.index
-            ) as remote:
-                await salobj.set_summary_state(
-                    remote=remote, state=salobj.State.ENABLED, settingsToApply="arrays",
-                )
+        async with self.make_remote_and_topic_writer() as remote:
+            await salobj.set_summary_state(
+                remote=remote, state=salobj.State.ENABLED, settingsToApply="arrays",
+            )
 
-                num_samples = 10000
+            num_samples = 1000
 
-                # Wait for the first sample so we know the writer is running.
+            # Wait for the first sample so we know the writer is running.
+            data = await remote.tel_arrays.next(flush=False, timeout=STD_TIMEOUT)
+            t0 = time.monotonic()
+            for i in range(num_samples):
                 data = await remote.tel_arrays.next(flush=False, timeout=STD_TIMEOUT)
-                t0 = time.monotonic()
-                for i in range(num_samples):
-                    data = await remote.tel_arrays.next(
-                        flush=False, timeout=STD_TIMEOUT
-                    )
-                dt = time.monotonic() - t0
-                arrays_read_speed = num_samples / dt
-                print(f"read {arrays_read_speed:0.0f} arrays samples/second")
-                self.assertTrue(data.int0[0], num_samples)
-
-                if verify is not None:
-                    self.insert_measurement(
-                        verify.Measurement(
-                            "salobj.ReadTest_arrays",
-                            arrays_read_speed * u.ct / u.second,
-                        )
-                    )
+            dt = time.monotonic() - t0
+            arrays_read_speed = num_samples / dt
+            nlost = data.int0[0] - num_samples
```

Comment:
This looks like it is comparing a fixed number in an array instead of something that's been counted to num_samples. If data.int0[0] = 1000, nlost is always 0. What am I missing?

Reply:
See my comment just above (in the conversation view). This is the same thing being measured as for the logLevel messages, but for a large topic (one with arrays, though I only set the first item in the int0 array).